packages/devtools/lib/src/utils.dart

```diff
@@ -344,12 +344,18 @@ enum TimeUnit {
 }
 
 class TimeRange {
+  TimeRange({this.immutable = true});
```

Comment:
this isn't really immutable. You  are allowing it to be mutated once. immutable would imply to me that you cannot set the values at all after treating the object.
Maybe `immutable` --> `singleAssignment` or another word that imples you can set it once?

Reply:
good point. fixed.